components/ui/tabcounter/src/main/java/org/mozilla/components/ui/tabcounter/TabCounter.kt

```diff
@@ -0,0 +1,368 @@
+        // The first animator, fadeout in 33 ms (49~51, 2 frames).
+        val fadeOut = ObjectAnimator.ofFloat(box, "alpha",
+                ANIM_BOX_FADEOUT_FROM, ANIM_BOX_FADEOUT_TO).setDuration(ANIM_BOX_FADEOUT_DURATION)
+
+        // Move up on y-axis, from 0.0 to -5.3 in 50ms, with fadeOut (49~52, 3 frames).
+        val moveUp1 = ObjectAnimator.ofFloat(box, "translationY",
+                ANIM_BOX_MOVEUP1_TO, ANIM_BOX_MOVEUP1_FROM).setDuration(ANIM_BOX_MOVEUP1_DURATION)
+
+        // Move down on y-axis, from -5.3 to -1.0 in 116ms, after moveUp1 (52~59, 7 frames).
+        val moveDown2 = ObjectAnimator.ofFloat(box, "translationY",
+                ANIM_BOX_MOVEDOWN2_FROM, ANIM_BOX_MOVEDOWN2_TO).setDuration(ANIM_BOX_MOVEDOWN2_DURATION)
+
+        // FadeIn in 66ms, with moveDown2 (52~56, 4 frames).
+        val fadeIn = ObjectAnimator.ofFloat(box, "alpha",
+                ANIM_BOX_FADEIN_FROM, ANIM_BOX_FADEIN_TO).setDuration(ANIM_BOX_FADEIN_DURATION)
+
+        // Move down on y-axis, from -1.0 to 2.7 in 116ms, after moveDown2 (59~66, 7 frames).
+        val moveDown3 = ObjectAnimator.ofFloat(box, "translationY",
+                ANIM_BOX_MOVEDOWN3_FROM, ANIM_BOX_MOVEDOWN3_TO).setDuration(ANIM_BOX_MOVEDOWN3_DURATION)
+
+        // Move up on y-axis, from 2.7 to 0 in 133ms, after moveDown3 (66~74, 8 frames).
+        val moveUp4 = ObjectAnimator.ofFloat(box, "translationY",
+                ANIM_BOX_MOVEDOWN4_FROM, ANIM_BOX_MOVEDOWN4_TO).setDuration(ANIM_BOX_MOVEDOWN4_DURATION)
+
+        // Scale up height from 2% to 105% in 100ms, after moveUp1 and delay 16ms (53~59, 6 frames).
+        val scaleUp1 = ObjectAnimator.ofFloat(box, "scaleY",
+                ANIM_BOX_SCALEUP1_FROM, ANIM_BOX_SCALEUP1_TO).setDuration(ANIM_BOX_SCALEUP1_DURATION)
+        scaleUp1.startDelay = ANIM_BOX_SCALEUP1_DELAY // delay 1 frame after moveUp1
+
+
+        // Scale down height from 105% to 99% in 116ms, after scaleUp1 (59~66, 7 frames).
+        val scaleDown2 = ObjectAnimator.ofFloat(box, "scaleY",
+                ANIM_BOX_SCALEDOWN2_FROM, ANIM_BOX_SCALEDOWN2_TO).setDuration(ANIM_BOX_SCALEDOWN2_DURATION)
+
+        // Scale up height from 99% to 100% in 133ms, after scaleDown2 (66~74, 8 frames).
+        val scaleUp3 = ObjectAnimator.ofFloat(box, "scaleY",
+                ANIM_BOX_SCALEUP3_FROM, ANIM_BOX_SCALEUP3_TO).setDuration(ANIM_BOX_SCALEUP3_DURATION)
+
+        animatorSet.play(fadeOut).with(moveUp1)
+        animatorSet.play(moveUp1).before(moveDown2)
+        animatorSet.play(moveDown2).with(fadeIn)
+        animatorSet.play(moveDown2).before(moveDown3)
+        animatorSet.play(moveDown3).before(moveUp4)
+
+        animatorSet.play(moveUp1).before(scaleUp1)
+        animatorSet.play(scaleUp1).before(scaleDown2)
+        animatorSet.play(scaleDown2).before(scaleUp3)
+    }
+
+    private fun createBarAnimatorSet(animatorSet: AnimatorSet) {
+        val firstAnimator = animatorSet.childAnimations[0]
+
+
```

Comment:
\[ktlint\] WARNING: \[no\-consecutive\-blank\-lines\] Needless blank line\(s\) in column 1